### PR TITLE
feat(go-sdk): add JSON support to the SDK

### PR DESCRIPTION
**Pull request by fourstepper**
Closes https://github.com/OpenSLO/OpenSLO/issues/283 

WIP right now:

1. Things that don't belong here are here
2. Neither https://github.com/OpenSLO/OpenSLO/blob/main/pkg/openslosdk/format.go or https://github.com/OpenSLO/OpenSLO/blob/main/pkg/openslosdk/encoding.go were touched yet

### CODE REVIEW

File: pkg/openslo/v1/sli.go [hunk 1]
```diff
@@ -30,25 +32,26 @@ func (s SLI) Validate() error {
 }
 
 type SLISpec struct {
-	ThresholdMetric *SLIMetricSpec `yaml:"thresholdMetric,omitempty"`
-	RatioMetric     *RatioMetric   `yaml:"ratioMetric,omitempty"`
+	Description     string         `yaml:"description,omitempty" json:"description,omitempty"`
```
Comment (fourstepper) [inline pkg/openslo/v1/sli.go:35]: added Description here

**Comment by fourstepper** [comment]
@nieomylnieja I'm looking at the https://github.com/OpenSLO/OpenSLO/blob/main/pkg/openslosdk/encoding.go and https://github.com/OpenSLO/OpenSLO/blob/main/pkg/openslosdk/format.go files and I don't really get what their purpose is? Is this just some functions that we've decided to provide from the SDK to the users of the SDK?

**Comment by nieomylnieja** [comment]
@fourstepper since OpenSLO generic object as defined by `openslo.Object` is an interface, we can't just use `json` or `yaml` unmarshal functions on it. Decoding into interface will yield an error as the operation is ambiguous. Thus, we need to provide utilities to decode any object's version into this slice of `openslo.Object`.

**Comment by nieomylnieja** [comment]
This PR will also resolve the issue of JSON support: https://github.com/OpenSLO/OpenSLO/pull/291. If we go with this library change we'll close this PR.

**Pull request closed without merging**